colossalai/auto_parallel/tensor_shard/utils/reshape.py

```diff
@@ -0,0 +1,168 @@
+from enum import Enum
+from typing import Dict, List, Tuple
+
+import torch
+
+
+class PreviousStatus(Enum):
+    """
+    This class shows the status of previous comparision.
+    """
+    RESET = 0
+    # ORIGIN means the origin element is larger in the previous comparision.
```

Comment:
What is `original element`?

Reply:
Do you mean tensor?

Reply:
The element belong to the origin tensor, and target element means the element belong to the reshaped tensor.

Reply:
Element means a value in a tensor?

Reply:
Element means a value in tensor.shape, sorry about this, I will add some comment in my code to explain this issue.

Reply:
should it be named as dimension_size?

Reply:
Sure, I will rename this variable.